qiskit/algorithms/state_fidelities/compute_uncompute.py

```diff
@@ -0,0 +1,145 @@
+
+        Returns:
+            The fidelity quantum circuit corresponding to circuit_1 and circuit_2.
+        """
+        circuit = circuit_1.compose(circuit_2.inverse())
+        circuit.measure_all()
+        return circuit
+
+    def _run(
+        self,
+        circuits_1: QuantumCircuit | Sequence[QuantumCircuit],
+        circuits_2: QuantumCircuit | Sequence[QuantumCircuit],
+        values_1: Sequence[float] | Sequence[Sequence[float]] | None = None,
+        values_2: Sequence[float] | Sequence[Sequence[float]] | None = None,
+        **run_options,
+    ) -> StateFidelityResult:
+        r"""
+        Computes the state overlap (fidelity) calculation between two
+        (parametrized) circuits (first and second) for a specific set of parameter
+        values (first and second) following the compute-uncompute method.
+
+        Args:
+            circuits_1: (Parametrized) quantum circuits preparing :math:`|\psi\rangle`.
+            circuits_2: (Parametrized) quantum circuits preparing :math:`|\phi\rangle`.
+            values_1: Numerical parameters to be bound to the first circuits.
+            values_2: Numerical parameters to be bound to the second circuits.
+            run_options: Backend runtime options used for circuit execution. The order
+                of priority is\: run_options in ``run`` method > fidelity's default
+                run_options > primitive's default setting.
+                Higher priority setting overrides lower priority setting.
+
+        Returns:
+            The result of the fidelity calculation.
+
+        Raises:
+            ValueError: At least one pair of circuits must be defined.
+            AlgorithmError: If the sampler job is not completed successfully.
+        """
+
+        circuits = self._construct_circuits(circuits_1, circuits_2)
+        if len(circuits) == 0:
+            raise ValueError(
+                "At least one pair of circuits must be defined to calculate the state overlap."
+            )
+        values = self._construct_value_list(circuits_1, circuits_2, values_1, values_2)
+
+        # The priority of run options is as follows:
+        # run_options in `evaluate` method > fidelity's default run_options >
+        # primitive's default run_options.
+        run_opts = copy(self._default_run_options)
+        run_opts.update(**run_options)
+
+        job = self._sampler.run(circuits=circuits, parameter_values=values, **run_opts)
+
+        try:
+            result = job.result()
+        except Exception as exc:
+            raise AlgorithmError("Sampler job failed!") from exc
+
+        raw_fidelities = [prob_dist.get(0, 0) for prob_dist in result.quasi_dists]
+        fidelities = self._truncate_fidelities(raw_fidelities)
+
+        return StateFidelityResult(
+            fidelities=fidelities,
+            raw_fidelities=raw_fidelities,
+            metadata=result.metadata,
+            run_options=run_opts,
```

Comment:
This is just a comment. In gradients, in the result run_options there it says of run_options

>run_options for the estimator. Currently, estimator's default run_options is not included.
        
which is the case here right. So these end up being the options we give to sampler.run() which may then augment them with it own default option.s If run_options were part of Sampler ( and Estimator) result then like metadata it could be result.run_options. But until such time that this happens, its just perhaps something to bear in mind.